### ISSUE TITLE
Request clean up is not not done in some cases.

### ISSUE THREAD

**Issue by sergey-gt-ekb** [FreeRADIUS/freeradius-server#2261]
# Issue type
- Defect - Unexpected behaviour (obvious or verified by project member).
Request cleanup is not not done in some cases.

# Defect
Request cleanup is not not done in some cases.

## How to reproduce the issue
Setup low max_request for test, for example 2.
Configure server to not send response via "do_not_respond" policy
Send several access-query with retransmits.
After cleanup time, cleanup not done and result: request cache overflowed.

## Output of ``[radiusd|freeradius] -X`` showing issue occurring
_(you may need to run ``[radiusd|freeradius] -fxx -l stdout`` if using eg RADIUS with TLS)_

```text
....
(2) Not responding to request
(2) Not sending reply
(2) Finished request
Waking up in 0.3 seconds.
Waking up in 4.6 seconds.   -- Its OK!
Received Access-Request Id 53 from 10.147.255.18:14197 to 10.66.4.55:1812 length 80
(2) No reply.  Ignoring retransmit
Waking up in 999995.0 seconds.   -- WTF???!!!
Received Access-Request Id 53 from 10.147.255.18:14197 to 10.66.4.55:1812 length 80
(2) No reply.  Ignoring retransmit
Waking up in 1999389.8 seconds.   -- increased by 1m seconds again! may be USEC constant added to seconds???
Received Access-Request Id 114 from 10.147.255.18:13537 to 10.66.4.55:1812 length 80
Dropping request (3 is too many): from client Monitoring port 13537 - ID: 114
Please check the configuration file.    The value for 'max_requests' is probably set too low. 
Waking up in 1999096.2 seconds.
-- no cleanup result: request cache overflowed!!!
```
## Full backtrace from LLDB or GDB

```text
COPY/PASTE OUTPUT HERE (WITHIN BACKTICKS).  NO PASTEBIN (ET AL) LINKS!
```

**Comment by alandekok** [comment]
> Waking up in 999995.0 seconds

Which looks like you're not using 3.0.17.  Please re-test with 3.0.17.

**Comment by sergey-gt-ekb** [comment]
You are right!

**Comment by alandekok** [comment]
We're not going to debug issues which were found and fixed years ago.

Try with 3.0.17.  If you can still reproduce the problem, then re-open the issue with new debug output.

**Comment by sergey-gt-ekb** [comment]
We are use centos 5 in production environment.  Latest versions has compatible issues with centos5.
I will try find out a commit with fix and patch 3.0.4.
